api/controllers/ApplicationTypeController.js

```diff
@@ -54,93 +55,87 @@ module.exports = {
-                    const standardAppCountQuery = 'SELECT count(*) FROM "Application" WHERE "user_id" =:userId and "serviceType" = 1 and "createdAt" > NOW() - INTERVAL \'' + sails.config.standardServiceRestrictions.appSubmissionTimeFrameInDays + ' days\' and ("submitted" =:submitted OR "submitted" =:queued)';
-
-                    return sequelize.query(
-                        standardAppCountQuery,
-                        { replacements: {userId: user.id, submitted: 'submitted', queued: 'queued'},
-                        type: sequelize.QueryTypes.SELECT
-                    }).then(() => {
-                        const userData = HelperService.getUserData(req, res);
-                        const serviceSelectorView = 'eApostilles/applicationType.ejs';
-
-                        req.session.user = user;
-                        req.session.account = account;
-                        req.session.appId = false; // reset the appId so a new session is used
-                        // set initial submit status to false, meaning it application has not yet been submitted
-                        req.session.appSubmittedStatus = false;
-                        req.session.email_sent = false;
-
-                        return res.view(serviceSelectorView, {
-                            application_id: 0,
-                            userServiceURL:
-                                sails.config.customURLs.userServiceURL,
-                            error_report: false,
-                            changing: false,
-                            form_values: false,
-                            submit_status: req.session.appSubmittedStatus,
-                            current_uri: req.originalUrl,
-                            user_data: userData,
-                            back_link: req.session.startBackLink,
-                            //disableStandardServiceSection: disableStandardServiceSection
-                        });
-                    });
-                });
-            });
-
-        }
-
         req.session.appId = false; // reset the appId so a new session is used
         // set initial submit status to false, meaning it application has not yet been submitted
         req.session.appSubmittedStatus = false;
 
-        return res.view('applicationForms/applicationType.ejs', {
+        const userModels = getUserModels(req._sails.config.userServiceSequelize);
+
+        return this._renderServiceSelectionPage(req, res, userModels);
+    },
+
+    _renderServiceSelectionPage(req, res, userModels) {
+        const userLoggedIn = HelperService.LoggedInStatus(req);
+        const userData = HelperService.getUserData(req, res);
+        const pageData = {
             application_id: 0,
             userServiceURL: sails.config.customURLs.userServiceURL,
-            error_report: false,
+            error_report: Boolean(req.query.error),
```

Comment:
This is the main bit that was changed, the rest is refactoring for testing reasons.

Reply:
should this be using post redirect get + flash?

Reply:
Yeah I guess this could be using a flash but I'm not sure about post redirect get? I guess I'm not quite sure what you mean. 😅